### PR TITLE
Feature/csp builder

### PR DESCRIPTION
**Pull request by robot-5**
Adds Builder to create CSP strings in a more safe/convenient way, validating CSP directives. It is relatively much code since a number of CSP directives have quite different requirements.
I would be grateful for any feedback :slightly_smiling_face:

### CODE REVIEW

**Review by unrolled — COMMENTED** [review]
This is awesome, thank you for taking the time to tackle this! I enabled the tests for the PR and see a few linting errors. I think this is good to merge once those are fixed up.

**Comment by robot-5** [comment]
The linting issues should now be fixed!

File: cspbuilder/builder.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+	if err != nil {
+		panic(err)
+	}
+	return policy
+}
+
+// Build creates a content security policy string from the specified directives.
+// If any directive contains invalid values, an error is returned instead.
+func (builder *Builder) Build() (string, error) {
+	var sb strings.Builder
```
Comment (unrolled) [inline cspbuilder/builder.go:62]: I really like how you used `strings.Builder` here 👍

**Review by unrolled — APPROVED** [review]
Looks great! (Resolves #73)

**Pull request merged**